MinecraftClient/ChatBots/MineCube.cs

```diff
@@ -0,0 +1,340 @@
+                        break;
+                    }
+                }
+            }
+            else
+            {
+                LogToConsole("Activate Inventory Handling.");
+            }
+        }
+
+        public bool IsSorroundedByGravityBlocks(Location block)
+        {
+            World world = GetWorld();
+            double blockX = Math.Round(block.X);
+            double blockY = Math.Round(block.Y);
+            double blockZ = Math.Round(block.Z);
+
+            List<Material> gravityBlockList = new List<Material>(new Material[] {Material.Gravel, Material.Sand, Material.RedSand, Material.Scaffolding, Material.Anvil, });
+            List<Material> liquidBlockList = new List<Material>(new Material[] { Material.Water, Material.Lava, });
+
+            var temptype = world.GetBlock(new Location(blockX, blockY + 1, blockZ)).Type;
+            var tempLoc = gravityBlockList.Contains(world.GetBlock(new Location(blockX, blockY + 1, blockZ)).Type);
+
+            return
+                // Block can not fall down on player e.g. Sand, Gravel etc.
+                gravityBlockList.Contains(world.GetBlock(new Location(blockX, blockY + 1, blockZ)).Type) ||
+                
+                // Liquid can not flow down the hole. Liquid is unable to flow diagonally.
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX, blockY + 1, blockZ)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX - 1, blockY, blockZ)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX + 1, blockY, blockZ)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX, blockY, blockZ - 1)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX, blockY, blockZ + 1)).Type);
+        }
+
+        /// <summary>
+        /// Prints a whole cube to the console. Separated in layers and rows.
+        /// </summary>
+        /// <param name="cubeToPrint">Some cube</param>
+        private void PrintCubeToConsole(Cube cubeToPrint)
+        {
+            LogToConsole("Cube generated:");
+            foreach (Layer lay in cubeToPrint.LayersToMine)
+            {
+                LogToConsole("Layer:");
+                foreach (Row r in lay.RowsToMine)
+                {
+                    string generatedRow = "Row: ";
+                    foreach (Location loc in r.BlocksToMine)
+                    {
+                        generatedRow += loc.ToString() + "; ";
+                    }
+                    LogToConsole(generatedRow);
+                }
+            }
+            LogToConsole("End of cube.");
+        }
+
+        private string EvaluateMineUpCommand(string command, string[] args)
+        {
+            Location startBlock = new Location(
+                        double.Parse(args[0]),
+                        double.Parse(args[1]),
+                        double.Parse(args[2])
```

Comment:
It is a good practice to validate user input before using them. You may take a look at how `AutoTree` does it.
https://github.com/ORelio/Minecraft-Console-Client/blob/349a3de8e4a5fb5d2af3fe9b74aa700e2de2a1a8/MinecraftClient/config/ChatBots/AutoTree.cs#L142-L144
https://github.com/ORelio/Minecraft-Console-Client/blob/349a3de8e4a5fb5d2af3fe9b74aa700e2de2a1a8/MinecraftClient/config/ChatBots/AutoTree.cs#L146-L158